### ISSUE TITLE
How to run MDGAT-matcher in other dataset

### ISSUE THREAD

**Issue by LimHyungTae** [chenghao-shi/MDGAT-matcher#10]
Dear Chengho, I'm wondering how to run MDGAT-matcher in other datasets. In KITTI, you've already provided previously extracted keypoints, yet in some cases, users have to extract their own keypoints.

I saw that in USIP repository, there's `save_keypoints.py`, but I'm not sure how to reproduce feature extraction that you provide. 

Could you give me some hints? What kind of pre-trained model do you guys use?

I'd like to adopt your algorithm as one of the baseline algorithms, and I'd like to run MDGAT-matcher in the MulRan dataset.

Thank you in advance :)

**Comment by chenghao-shi** [comment]
Hi lim,

The provided keypoints are 37-dimensional features formed as [x, y, z, score, 33-dim FPFH]. If you wish to test on other datasets, you can extract the keypoints and generate the FPFH feature based on them, then combine the keypoints with features. It is not necessary to have a score value (you can assign a constant value if one is not available), since the score is only used to filter keypoints. Note that if you want to use USIP, you need to preprocess the datasets to calculate normal values for each point. We used the pretrained model on KITTI, by the way.

Actually, I had extracted the keypoints on MulRan, do you have a way to access Baidu Netdisk?

**Comment by LimHyungTae** [comment]
I tried to sign up, but I failed because I have no Chinese phone number :( 

But I can access your `pan.baidu.com` address, i.e., [this link](https://pan.baidu.com/s/1XJ-1YBN6kAiAud9xQg2B4w ). 

Could you please share your pre-extracted keypoints? I'm on the revision and it would be a great help for me! Thank you in advance.

**Comment by chenghao-shi** [comment]
Hi

Here is the [link](https://pan.baidu.com/s/13fM8Ed1HCtAfkDYpxfGBVA) to keypoints extracted from Mulran, code:xkzr.
Hope you doing well with your revision :)

**Comment by LimHyungTae** [comment]
Oops, I didn't notice that download is only possible if I have an baidu account, but I failed :(

Could you send me that file via e-mail (`shapelim@kaist.ac.kr`) or google drive? 

Sorry for bothering you ㅜㅠ, but please help me one more time

**Comment by karim416** [comment]
Dear Chenghao ,

Thank you for this very interesting work. I tried yout method on ModelNet 40, as you suggested i assigned a constant value for the keypont score. However, i obtain values that under are my expectation for matching metrics. Do you have any tip to try improvingn the performances of MDGAT for noisy data ?


Thank you again.
